### PR TITLE
fix(makeServerFetchye): mask errors from server side fetching

### PR DESCRIPTION
**Pull request by code-forger**
When a fetch fails in SSR, we now drop the error entirely. This prevents unserializable Errors ending up in the redux store. 

This also provides a better behaviour for SSR where failed fetches still render a loading state, allowing the browser to 'try again' fetching the data

## Motivation and Context
This removes a 'cannot serialize' error from being emitted on the server when a fetch fails.

## How Has This Been Tested?
Using a local instance of one app and some test modules I have proved that SSR now renders a 'loading' state of the application, without logging 'cannot serialize' errors, when a fetch fails in the server.
I have Unit test coverage for all new code.

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

## Checklist:

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] My changes are in sync with the code style of this project.
- [x] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [x] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using Fetchye?
None

### CODE REVIEW

**Review by JAdshead — APPROVED** [review]
I am happy to always have another attempt in the client if it errors on the server.

**Review by 10xLaCroixDrinker — CHANGES_REQUESTED** [review]

File: packages/fetchye-core/__tests__/ssrFetcher.spec.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
```
[the remark this answers is not in the document]
Comment (code-forger) [reply packages/fetchye-core/__tests__/ssrFetcher.spec.js]: address in 2e59af6

File: packages/fetchye-core/src/ssrFetcher.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
```
[the remark this answers is not in the document]
Comment (code-forger) [reply packages/fetchye-core/src/ssrFetcher.js]: address in 2e59af6

File: packages/fetchye-core/__tests__/ssrFetcher.spec.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+describe('ssrFetcher', () => {
+  it('forward all params to the default fetcher, and return the payload from the default fetcher with error: null', async () => {
+    const response = await ssrFetcher('paramMock1', 'paramMock2', 'paramMock3');
+    expect(response).toEqual({
+      payload: 'payloadMock',
+      error: null,
+    });
+
+    expect(defaultFetcher).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (code-forger) [reply packages/fetchye-core/__tests__/ssrFetcher.spec.js]: It was validating that the parameters passed to ssrFetcher were forwarded to defaultFetcher.
address in 2e59af6

**Review by JAdshead — APPROVED** [review]

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Pull request merged**